packages/twenty-front/src/modules/workflow/workflow-steps/workflow-actions/code-action/components/WorkflowEditActionServerlessFunction.tsx

```diff
@@ -289,6 +388,116 @@ export const WorkflowEditActionServerlessFunction = ({
 
   const testLogsTextAreaId = `${serverlessFunctionId}-test-logs`;
 
+  const getCodeEditorOptions = (includeLineNumbersConfig = true) => ({
+    readOnly: actionOptions.readonly,
+    domReadOnly: actionOptions.readonly,
+    scrollBeyondLastLine: false,
+    padding: { top: 4, bottom: 4 },
+    ...(includeLineNumbersConfig && { lineNumbersMinChars: 2 }),
+  });
+
+  const handleEnterFullScreen = () => {
+    setIsFullScreen(true);
+    // Focus the overlay after it's rendered
+    setTimeout(() => {
+      if (isDefined(fullScreenOverlayRef.current)) {
+        fullScreenOverlayRef.current.focus();
+      }
+    }, 0);
+  };
+
+  const handleExitFullScreen = () => {
+    setIsFullScreen(false);
+  };
+
+  const breadcrumbLinks: BreadcrumbProps['links'] = [
+    {
+      children: workflow?.name?.trim() || t`Untitled Workflow`,
+      href: '#',
+    },
+    {
+      children: headerTitle,
+      href: '#',
+    },
+    {
+      children: t`Code Editor`,
+    },
+  ];
+
+  const renderCodeEditor = (fullScreenMode = false) => (
```

Comment:
we don't do this kind of pattern in the code, let's declare a component in another file (we usually avoid it because it makes long file and leads to refactoring when we want to re-use, if it's already a component devs are most likely to re-use)

Reply:
Ok!